File: python/sync/5_create_report.py
"""
5_create_report.py (sync)  —  STEP 5: Request an axes4 score report
===================================================================
Asks the API to generate an axes4 accessibility score report for a file.

EDIT NOTHING HERE. Set this in  ../config.json  under "report":
  "report": { "file_id": "<the file_id to generate a report for>" }

How to run:  python 5_create_report.py

What it saves to data.json:
  "report_process": [ {"file_id": "....", "job_id": "....", "status": "Processing"}, ... ]
"""

import requests
from helper import BASE_URL, load_config, api_key, build_headers, get_value, save_value, show_response, log_file_error

cfg = load_config()
key = api_key()
report = cfg.get("report") or {}
FILE_ID = str(report.get("file_id", "")).strip()

if not FILE_ID:
    print('[X] No file_id given. Set "report": {"file_id": ...} in config.json.')
    raise SystemExit

ENDPOINT = f"{BASE_URL}/report/"
payload = {"file_id": FILE_ID}

print(f"Requesting a score report for file_id {FILE_ID} ...")
response = requests.post(ENDPOINT, headers=build_headers(key), json=payload)
show_response(response)

if response.status_code in (200, 201):
    body = response.json()
    data = body.get("data") or {}
    job_id = data.get("job_id")

    if job_id:
        report_process = get_value("report_process", [])
        if not any(r.get("job_id") == job_id for r in report_process):
            report_process.append({"file_id": FILE_ID, "job_id": job_id, "status": "Processing"})
            save_value("report_process", report_process)
        print("\n[OK] Got report job_id:", job_id)
        print("Next: run  python 6_check_report.py")
    else:
        print("\n[!] Could not find 'job_id' in the response. "
              "Check the printed response above and update the key name.")
        log_file_error(FILE_ID, response.status_code, "No job_id in report-create response", body)
else:
    print("\n[X] Could not request the report. Check the file_id and status code above.")
    try:
        raw = response.json()
    except ValueError:
        raw = None
    log_file_error(FILE_ID, response.status_code, "Could not request report", raw)

File: node/5_create_report.js
/*
 * 5_create_report.js  —  STEP 5: Request an axes4 score report
 * ============================================================
 * Asks the API to generate an axes4 accessibility score report for a file.
 * Returns a report job_id.
 *
 * EDIT NOTHING HERE. Set this in  ../config.json  under "report":
 *   "report": { "file_id": "<the file_id to generate a report for>" }
 *
 * How to run:  node 5_create_report.js
 *
 * What it saves to data.json:
 *   "report_process": [ { "file_id": "....", "job_id": "....", "status": "Processing" }, ... ]
 */

const { BASE_URL, loadConfig, apiKey, buildHeaders, getValue, saveValue, showResponse, logFileError } = require("./helper");

async function main() {
  const cfg = loadConfig();
  const key = apiKey();
  const report = cfg.report || {};
  const fileId = (report.file_id || "").trim();

  if (!fileId) {
    console.log('[X] No file_id given. Set "report": {"file_id": ...} in config.json.');
    return;
  }

  const payload = { file_id: fileId };

  console.log(`Requesting a score report for file_id ${fileId} ...`);
  const response = await fetch(`${BASE_URL}/report/`, {
    method: "POST",
    headers: buildHeaders(key),
    body: JSON.stringify(payload),
  });

  const body = await showResponse(response);

  if ((response.status === 200 || response.status === 201) && body) {
    const data = body.data || {};
    const jobId = data.job_id;

    if (jobId) {
      const reportProcess = getValue("report_process", []);
      if (!reportProcess.some((r) => r.job_id === jobId)) {
        reportProcess.push({ file_id: fileId, job_id: jobId, status: "Processing" });
        saveValue("report_process", reportProcess);
      }
      console.log("\n[OK] Got report job_id:", jobId);
      console.log("Next: run  node 6_check_report.js");
    } else {
      console.log(
        "\n[!] Could not find 'job_id' in the response. " +
          "Check the printed response above and update the key name."
      );
      logFileError(fileId, response.status, "No job_id in report-create response", body);
    }
  } else {
    console.log(
      "\n[X] Could not request the report. Check the file_id and status code above."
    );
    logFileError(fileId, response.status, "Could not request report", body);
  }
}

main();
